### PR TITLE
Add functional.relu/gelu/softmax opinfos

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61068 Add functional.dropout opinfo
* **#61067 Add functional.relu/gelu/softmax opinfos**

### CODE REVIEW

**Comment by mruberry** [comment]
Test failures look real:

```
The supported backward dtypes for nn.functional.relu on cpu according to its OpInfo are
        {torch.float32, torch.float64}, but the detected supported backward dtypes are {torch.float32, torch.bfloat16, torch.float64}.
        The following backward dtypes should be added to the OpInfo: {torch.bfloat16}. 
```

```
The supported dtypes for softmax on cpu according to its OpInfo are
        {torch.float32, torch.bfloat16, torch.float64}, but the detected supported dtypes are {torch.float32, torch.float64}.
        The following dtypes should be removed from the OpInfo: {torch.bfloat16}.
```

```
The supported backward dtypes for nn.functional.relu on cuda according to its OpInfo are
        {torch.float16, torch.float64, torch.float32}, but the detected supported backward dtypes are {torch.float16, torch.bfloat16, torch.float64, torch.float32}.
        The following backward dtypes should be added to the OpInfo: {torch.bfloat16}. 
```

Let me know if those error messages makes sense. Do you get different results running the tests locally? Also: the way the test is written you may not get all the dtype errors upfront. Maybe we should change it to report all errors?

The NNC test test_unsupported_nn_functional_relu_cpu_float32 (__main__.TestNNCOpInfoCPU) is also failing, which we should move into test_ops.py and add metadata for NNC support so we don't need to edit a different file, and so python test_ops.py -k foo will run most op tests for foo. That should make it easier to write the OpInfo, run the test command, and then fix the metadata.

**Comment by mruberry** [comment]
These OpInfos have been added since this PR was opened
